src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,376 @@
+  return <Box sx={merge<BetterSystemStyleObject>({display: isVisible ? 'flex' : 'none'}, sx)}>{children}</Box>
+}
+
+// ContextAreaActions
+// ---------------------------------------------------------------------
+const ContextAreaActions: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  visible = onlyVisibleOnNarrowView
+}) => {
+  const isVisible = useResponsiveValue(visible, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {
+          display: isVisible ? 'flex' : 'none',
+          flexDirection: 'row',
+          alignItems: 'center',
+          gap: '8px',
+          flexGrow: '1',
+          justifyContent: 'right'
+        },
+        sx
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+// PageHeader.TitleArea: The main title area of the page. Visible on all viewports.
+// PageHeader.TitleArea Sub Components: PageHeader.LeadingAction, PageHeader.LeadingVisual, PageHeader.Title, PageTitle.TrailingVisual, PageHeader.TrailingAction, PageHeader.Actions
+// PageHeader.LeadingAction and PageHeader.TrailingAction are only visible on regular viewports therefore they come as visible on narrow viewports and their visibility can be managed by their exposed `visible` prop
+// ---------------------------------------------------------------------
+const TitleArea: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, visible = true}) => {
+  const isVisible = useResponsiveValue(visible, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {display: isVisible ? 'flex' : 'none', flexDirection: 'row', alignItems: 'center', gap: '8px'},
+        sx
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+const LeadingAction: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  visible = visibleOnRegularView
+}) => {
+  const isVisible = useResponsiveValue(visible, true)
+  return <Box sx={merge<BetterSystemStyleObject>({display: isVisible ? 'flex' : 'none'}, sx)}>{children}</Box>
+}
+
+const LeadingVisual: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  visible = {
+    narrow: true,
+    regular: false,
+    wide: false
+  }
+}) => {
+  const isVisible = useResponsiveValue(visible, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {
+          display: isVisible ? 'flex' : 'none'
+        },
+        sx
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+export type TitleProps = {
+  variant?: 'subtitle' | 'medium' | 'large' | ResponsiveValue<'subtitle' | 'medium' | 'large'>
+  // Check if we need responsive values for heading is so should we update as prop's type for Heading component?
+  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+} & PageHeaderProps
+
+const Title: React.FC<React.PropsWithChildren<TitleProps>> = ({
+  children,
+  sx = {},
+  variant = 'medium',
+  visible = true,
+  as = 'h3'
```

Comment:
When would you say we should use the `as` prop versus something like:

```
<PageHeader.Title>
  <h3>Pull Request</h3>
</PageHeader.Title>
```

That lets them compose through `children`? Would it be that we want it as a prop because it _must_ be a heading of a specific level? I also think we should be careful about default heading levels and that it might help to make the level explicit instead of potentially providing an inappropriate default.

Reply:
I like that approach! I completely agree on being careful about the default level and considering cases where the title is actually not a title (see this [example](https://github.com/primer/react/pull/2265#discussion_r1029967215)), we could benefit from this approach. Thanks!

Reply:
I'll tackle this on a different PR @joshblack , if that is okay? I created an [issue](https://github.com/github/primer/issues/1571) for it. Let me know if you have any concerns.